src/java.base/share/classes/java/lang/Shutdown.java

```diff
@@ -157,16 +157,35 @@ static void halt(int status) {
      * which should pass a nonzero status code.
      */
     static void exit(int status) {
+        System.Logger log = getRuntimeExitLogger(); // Locate the logger without holding the lock;
         synchronized (Shutdown.class) {
             /* Synchronize on the class object, causing any other thread
              * that attempts to initiate shutdown to stall indefinitely
              */
+            if (log != null) {
+                Throwable throwable = new Throwable("Runtime.exit(" + status + ")");
+                log.log(System.Logger.Level.DEBUG, "Runtime.exit() called with status: " + status,
+                        throwable);
+            }
             beforeHalt();
             runHooks();
             halt(status);
         }
     }
 
+    // Locate and return the logger for Shutdown.exit, if it is functional and DEBUG enabled.
+    // Exceptions should not prevent System.exit; the exception is printed and otherwise ignored.
+    private static System.Logger getRuntimeExitLogger() {
```

Comment:
The method descriptions in this this class use /* .. */ style so probably best to keep it consistent if you.